source/common/router/router_ratelimit.h

```diff
@@ -129,8 +128,11 @@ class RateLimitPolicyImpl : public RateLimitPolicy {
   getApplicableRateLimit(int64_t stage = 0) const override;
 
 private:
-  std::vector<std::vector<std::unique_ptr<RateLimitPolicyEntry>>> rate_limit_entries_;
-  std::vector<std::vector<std::reference_wrapper<const RateLimitPolicyEntry>>>
+  std::vector<std::unique_ptr<RateLimitPolicyEntry>> rate_limit_entries_;
+  std::vector<std::reference_wrapper<const RateLimitPolicyEntry>>
+      default_rate_limit_entries_reference_;
+  // Holds reference to rate limit policies applicable to non-default stage settings.
+  std::unordered_map<int64_t, std::vector<std::reference_wrapper<const RateLimitPolicyEntry>>>
```

Comment:
any reason to have both default_rate_limit_entries_reference_ and rate_limit_entries_reference_, but not just rate_limit_entries_reference_ and default is stored with the key 0?